dns/client.go

```diff
@@ -55,6 +74,33 @@ func (r *Client) Resolve(
 	query, recordType string,
 	nameserver Nameserver,
 ) ([]string, error) {
+	// Ensure k6 client is initialized (lazy initialization)
+	r.ensureK6Client()
+
+	vustate := r.vu.State()
+	if vustate == nil {
+		return nil, fmt.Errorf("failed to get vu state")
+	}
+
+	// If k6's dialer is available, honor its blocked hostnames configuration.
+	//
+	// As the dns resolution process involves querying a specific nameserver, via its IP address,
+	// to query about a specific hostname, we need to check if the hostname is blocked by the k6 dialer.
+	//
+	// As such, we explicitly perform a check against the DNS query, and can't rely on the underlying dialer's
+	// existing logic that performs the check on the actual IP address targeted by the connection (which in our
+	// case is the nameserver's IP address, not the hostname).
+	if d, ok := vustate.Dialer.(*netext.Dialer); ok && d != nil && d.BlockedHostnames != nil {
```

Comment:
I'm not getting why the dialer could not be available. This is a k6 primitive, so not having it available means there is a critical issue. Did I miss something?